test/integration/testPageload.js

```diff
@@ -5,49 +5,64 @@ module.exports = {
     'Pageload, normal': function (browser) {
         var chai = require('chai');
         var uuidv4pattern = /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/;
-        var noUserIdResult = {};
         browser
-        .url('http://127.0.0.1:8080/dev/index.html')
-        .waitForElementVisible('body', 1000)
 		.deleteCookies()
         .getCookies(function callback(result) {
             this.assert.equal(result.value.length, 0);
         })
+        .url('http://127.0.0.1:8080/dev/index.html')
+        .waitForElementVisible('body', 1000)
         .pause(3000)
         .getCookies(function callback(result) {
 
-            noUserIdResult = result;
-            this.assert.equal(result.value.length, 4);
-            this.assert.equal(result.value[0].name, '_DataTrackerSession');
-            chai.assert.match(result.value[0].value, uuidv4pattern);
-            this.assert.equal(result.value[1].name, '_DataTrackerEnv');
-            chai.assert.isDefined(result.value[1].value, 'Environment ID was defined in cookie');
-            this.assert.equal(result.value[2].name, '_DataTrackerVisitor');
-            chai.assert.match(result.value[2].value, uuidv4pattern);
-            this.assert.equal(result.value[3].name, '_DataTrackerUser');
-            chai.assert.equal(result.value[3].value, 'undefined');
+			this.assert.equal(result.value.length, 4);
+
+			var count = 0;
+
+			for (var j = 0; j < result.value.length; j++) {
+				if (result.value[j].name === '_DataTrackerSession') {
```

Comment:
having if statements in tests makes me a little nervous - what are you testing here?